.github/workflows/lint.yml

```
@@ -0,0 +1,80 @@
+name: Lint
+
+on: [pull_request]
+
+env:
+  PLUGIN_NAME: dashboards-visualizations
+  OPENSEARCH_DASHBOARDS_VERSION: "main"
+
+jobs:
+  build:
+    name: Lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Plugin
+        uses: actions/checkout@v2
+
+      - name: Checkout OpenSearch Dashboards
+        uses: actions/checkout@v2
+        with:
+          repository: opensearch-project/Opensearch-Dashboards
+          ref: ${{ env.OPENSEARCH_DASHBOARDS_VERSION }}
+          path: OpenSearch-Dashboards
+
+      - name: Install Gantt Chart to Opensearch Dashboards Plugins Dir
+        run: |
+          mv gantt-chart ./OpenSearch-Dashboards/plugins
+
+      - name: Get node and yarn versions
+        working-directory: ${{ env.WORKING_DIR }}
+        id: versions_step
+        run: |
+          echo "::set-output name=node_version::$(cat ./OpenSearch-Dashboards/.nvmrc | cut -d"." -f1)"
+          echo "::set-output name=yarn_version::$(node -p "(require('./OpenSearch-Dashboards/package.json').engines.yarn).match(/[.0-9]+/)[0]")"
+
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ steps.versions_step.outputs.node_version }}
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Install correct yarn version for OpenSearch Dashboards
+        run: |
+          npm uninstall -g yarn
+          echo "Installing yarn ${{ steps.versions_step.outputs.yarn_version }}"
+          npm i -g yarn@${{ steps.versions_step.outputs.yarn_version }}
+
+      - name: Bootstrap the plugin
+        working-directory: OpenSearch-Dashboards/plugins/gantt-chart
+        run: yarn osd bootstrap
+
+      - name: Get list of changed files using GitHub Action
+        uses: lots0logs/gh-action-get-changed-files@2.2.2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Check Changes of Files
+        run: |
+          echo "FILES_MODIFIED=$(cat ${HOME}/files_modified.json)"
+          echo "FILES_ADDED=$(cat ${HOME}/files_added.json)"
+          echo "FILES_RENAMED=$(cat ${HOME}/files_renamed.json)"
+          echo "FILES_DELETED=$(cat ${HOME}/files_deleted.json)"
+
+      - name: Lint Changed Files
+        run: |
+          jq -r '.[]' ${HOME}/files_modified.json ${HOME}/files_added.json | sort | uniq > /tmp/changed_files.txt
+          CHANGED_FILES=$(cat /tmp/changed_files.txt)
+          echo "These are the changed files: $CHANGED_FILES"
+          if [[ -n "$CHANGED_FILES" ]]; then
+            echo "Linting changed files..."
+            while IFS= read -r file; do
+              if [[ $file == *.js || $file == *.ts || $file == *.tsx ]]; then
+                echo "linting file $file"
+                yarn lint "../$file"
+              fi
```

this could break if there's a `.ts` file outside of `gantt-chart`, because it will be using gantt-chart's eslint config. maybe
```suggestion
              if [[ $file =~ ^gantt-chart/.+\.(js|ts|tsx)$ ]]; then
                echo "linting file $file"
                yarn lint "../$file"
              fi
```

---

Thanks @joshuali925. That makes more sense. I just applied that.